src/wallet/redux/middleware.js

```diff
@@ -178,69 +178,61 @@ function connectFortmatic(store) {
+        break
+      case 'trust':
+        isAvailable = !!window.web3.currentProvider.isTrust
+        break
+      case 'cipher':
+        isAvailable = window.web3.currentProvider.constructor.name === 'CipherProvider'
+        break
+      case 'status':
+        isAvailable = !!window.web3.currentProvider.isStatus
+        break
+      case 'imtoken':
+        isAvailable = !!window.imToken
+        break
+      case 'coinbase':
+        isAvailable = !!window.web3.currentProvider.isToshi
+        break
+      case 'opera':
+        isAvailable =
+          ((!!window.opr && !!window.opr.addons) || !!window.opera || navigator.userAgent.indexOf(' OPR/') >= 0) &&
+          window.web3 &&
+          window.web3.currentProvider &&
+          window.web3.currentProvider.isConnected()
+        break
+      case 'equal':
+        isAvailable = !!window.web3.currentProvider.isEQLWallet
+        break
+      default:
+        isAvailable = false
     }
-
-    const walletsAvailable = {}
-    web3WalletTypes.map(type => {
-      let isAvailable = false
-      switch (type) {
-        case 'metamask':
-          isAvailable =
-            !!window.web3.currentProvider.isMetaMask && !isMobile.any && !window.web3.currentProvider.isEQLWallet
-          break
-        case 'trust':
-          isAvailable = !!window.web3.currentProvider.isTrust
-          break
-        case 'cipher':
-          isAvailable = window.web3.currentProvider.constructor.name === 'CipherProvider'
-          break
-        case 'status':
-          isAvailable = !!window.web3.currentProvider.isStatus
-          break
-        case 'imtoken':
-          isAvailable = !!window.imToken
-          break
-        case 'coinbase':
-          isAvailable = !!window.web3.currentProvider.isToshi
-          break
-        case 'opera':
-          isAvailable =
-            ((!!window.opr && !!window.opr.addons) || !!window.opera || navigator.userAgent.indexOf(' OPR/') >= 0) &&
-            window.web3 &&
-            window.web3.currentProvider &&
-            window.web3.currentProvider.isConnected()
-          break
-        case 'equal':
-          isAvailable = !!window.web3.currentProvider.isEQLWallet
-          break
-        default:
-          isAvailable = false
-      }
-      walletsAvailable[type] = isAvailable
-      return walletsAvailable
+    walletsAvailable[type] = isAvailable
+    return walletsAvailable
+  })
+  if (!_.isEqual(prevWalletsAvailable, walletsAvailable)) {
+    store.dispatch({
+      type: 'SET_WALLET_AVAILABILITY',
+      wallets: walletsAvailable,
     })
-    if (!_.isEqual(prevWalletsAvailable, walletsAvailable)) {
-      store.dispatch({
-        type: 'SET_WALLET_AVAILABILITY',
-        wallets: walletsAvailable,
-      })
-    }
   }
 }
 
 export default function walletMiddleware(store) {
   detectWeb3Wallets(store)
+  window.setInterval(() => detectWeb3Wallets(store), 5000)
```

Comment:
Do you need to clear the interval once it finds the web3 wallet?

Reply:
No the idea is to keep liveness, I also want to disable their ability to connect if they disable the wallet after they find it. The state could also be used in the future to prompt a window like `web3 connection lost, please re-enable your wallet` if the user disabled their extension in the middle of a session, or if the extension crashed for some reason.

Reply:
I thought maybe this was intentional in case the user disabled/re-enables the extension or something

Reply:
yep correct @perich

Reply:
Ah that makes sense 👍